File: tools/improve/tests/gemini-3-flash/maudec/cpp/euclid.cpp
//
//	<comment from="gemini-3-flash-preview">
//	The implementation was converted from a recursive to an iterative approach. This version improves efficiency by eliminating function call overhead and preventing potential stack overflow issues, while strictly preserving the original logic and signature of the Euclidean algorithm.
//	</comment>
//

int u0024gcd(int M, int N) {
    while (N != 0) {
        int temp = N;
        N = M % N;
        M = temp;
    }
    return M;
}

File: tools/improve/tests/gemini-3-flash/maudec/rust/euclid.rs
//
//	<comment from="gemini-3-flash-preview">
//	The implementation was optimized by replacing the recursive Euclidean algorithm with an iterative while loop. This change improves efficiency by eliminating the overhead of recursive function calls and avoiding potential stack overflow risks, while strictly preserving the mathematical semantics and the function signature of the original code.
//	</comment>
//

fn u0024gcd(mut m: i32, mut n: i32) -> i32 {
    while n != 0 {
        let temp = n;
        n = m % n;
        m = temp;
    }
    m
}